test/functional/p2p_fingerprint.py

```diff
@@ -121,14 +108,18 @@ def run_test(self):
         node0.sync_with_ping()
 
         # Request for very old stale block should now fail
+        with p2p_lock:
+            node0.last_message.pop("block", None)
         self.send_block_request(stale_hash, node0)
         time.sleep(3)
```

Comment:
unrelated: why not sync_with_ping?

Reply:
Good idea, done. It's always nice to get rid of time.sleep(...) constructs that are not really needed.